sql-db/hibernate-reactive/src/main/java/io/quarkus/ts/reactive/Library.java

```diff
@@ -0,0 +1,113 @@
+package io.quarkus.ts.reactive;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@Path("/library")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class Library {
+
+    @Inject
+    AuthorRepository authors;
+
+    @GET
+    @Path("books")
+    public Multi<String> all() {
+        return Book.all()
+                .map(Book::getTitle);
+    }
+
+    @GET
+    @Path("books/{id}")
+    public Uni<Response> find(Integer id) {
+        return Book.byId(id)
+                .map(book -> book == null
+                        ? Response.status(Response.Status.NOT_FOUND)
+                        : Response.ok(book.getTitle()))
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @PUT
+    @Path("books/{author}/{name}")
+    public Uni<Response> createBook(Integer author, String name) {
+        Book book = new Book();
+        book.setAuthor(author);
+        book.setTitle(name);
+        return Book.create(author, name)
+                .map(nothing -> Response.status(Response.Status.CREATED))
+                .onFailure().recoverWithItem(error -> Response.status(Response.Status.BAD_REQUEST).entity(error.getMessage()))
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @GET
+    @Path("books/author/{name}")
+    public Multi<String> search(String name) {
+        return authors.byName(name)
+                .flatMap(author -> Multi.createFrom().iterable(author.getBooks()))
```

Comment:
I prefer to use `.onItem.transform(...` instead of Maps or flapmaps
Have a look at methods as `transformToMulti`, `transformToIterable`...

For example in this case we are returning a list of titles, so why not do something like: 

```
static List<String> toBookList(Author author) {
        return author.getBooks().stream().map(b -> b.getTitle()).collect(Collectors.toList());
    }
    
     @GET
    @Path("books/author/{name}")
    public Multi<String> search(String name) {
        return authors.byName(name)
                .onItem()
                .transformToIterable(Library::toBookList);
}
```

And then ...I think that this helper method that I have created (`toBookList`) should be part of Author or book domain, not part of the endpoint (from a conceptual point of view). Is like...if you have a method `toJson` or `toString` is part of the object that is going to be serialized.

Reply:
Well, onItem() and friends are more verbose and aren't used anywhere outside Mutiny.
Map and flatmap, on the other side, are much more compact and very common both in JVM world[1],[2] and outside of it [3], [4].

Good idea about putting this into helper method, I changed that.

[1] https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html
[2] https://spark.apache.org/docs/latest/api/java/index.html?org/apache/spark/sql/Dataset.html
[3] https://doc.rust-lang.org/std/iter/trait.Iterator.html
[4] https://hexdocs.pm/elixir/Enum.html